hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -388,6 +502,29 @@ void GenericSystem::initialize_storage_vectors(
   }
 }
 
+// This method will populate the GPIO interface list, as there no standard interfaces
+void GenericSystem::populate_gpio_interfaces()
+{
+  std::set<std::string> interfaces_set;
```

Comment:
should we use set or rather check as for other interfaces?

Reply:
I think we can leave it with 'set' as this will be more clear. Also we can extend this function to generate the interfaces for any type of tag.